server/services/player.ts

```diff
@@ -0,0 +1,99 @@
+import { Request } from 'express';
+import { Match, Player, User } from '../models';
+import updatedPLayerSchema from '../validations/playerSchema';
+
+const getPlayerService = async (
+  id: number,
+): Promise<{
+  status: number;
+  data: string | object;
+}> => {
+  const player = await User.findOne({
+    where: { id },
+    attributes: ['username', 'phone', 'id'],
+    include: [
+      {
+        model: Player,
+      },
+    ],
+  });
+  if (!player) {
+    return {
+      status: 401,
+      data: 'هذا اللاعب غير موجود',
+    };
+  }
+
+  return {
+    status: 200,
+    data: player,
+  };
+};
+
+const updatePlayerService = async (
+  req: Request,
+): Promise<{
+  status: number;
+  data: string | object;
+}> => {
+  // const { UserId } = req.UserData;
+
+  const playerId = 5;
+  const { body } = req;
+  const isPLayerExist = await Player.findOne({ where: { UserId: playerId } });
+
+  if (!isPLayerExist) {
+    return {
+      status: 401,
+      data: 'هذا اللاعب غير متاح',
+    };
+  }
+  await updatedPLayerSchema.validateAsync(body);
```

Comment:
why you added all of the body you must add specific data

Reply:
how specific data if i don't know what is modifeied?